mozilla_schema_generator/main_ping.py

```diff
@@ -31,17 +31,46 @@ def get_schema(self):
         return self._update_env(schema)
 
     def _update_env(self, schema):
-        user_pref_elem = {
+        integer = {"type": "integer"}
+        string = {"type": "string"}
+        string_map = {
+            "type": "object",
+            "additionalProperties": string,
+        }
+        nested_string_map = {
+            "type": "object",
+            "additionalProperties": string_map,
+        }
+        map_without_values = {
             "type": "object",
             "additionalProperties": {
-                "type": "string"
-            }
+                "type": "object",
+            },
         }
 
+        active_addons = prepend_properties(("environment", "addons", "activeAddons")) + ("additionalProperties", "properties")
+        simple_measurements = prepend_properties(("payload", "simpleMeasurements", ""))[:-1]
+        processes_dynamic = prepend_properties(("payload","processes", "dynamic", ""))[:-1]
+
+        schema.set_schema_elem(
+                prepend_properties(("environment", "settings", "userPrefs")), string_map)
+        schema.set_schema_elem(
+                prepend_properties(("environment", "system", "os", "version")), string)
         schema.set_schema_elem(
-                prepend_properties(("environment", "settings", "userPrefs")), user_pref_elem)
+                prepend_properties(("environment", "addons", "theme", "foreignInstall")), integer)
         schema.set_schema_elem(
-                prepend_properties(("environment", "system", "os", "version")), {"type": "string"})
+                prepend_properties(("payload", "addonDetails", "XPI")), map_without_values)
+        schema.set_schema_elem(active_addons + ("foreignInstall",), integer)
+        schema.set_schema_elem(active_addons + ("version",), string)
+        schema.set_schema_elem(active_addons + ("userDisabled",), integer)
+        schema.set_schema_elem(processes_dynamic + ("scalars",), string_map)
+        schema.set_schema_elem(processes_dynamic + ("keyedScalars",), nested_string_map)
+        schema.set_schema_elem(simple_measurements + ("activeTicks",), integer)
```

Comment:
> I think we should explicitly set simple measures in `mozilla-pipeline-schemas`, because it is closer to the source of truth, and will actually get used for validation.

I would prefer to make the lowest-risk change that allows main summary and downstream stuff to not need `additional_properties`. So for now I don't want these to get used for validation, instead allowing anything and extracting these values only when present and correctly typed.

Reply:
Okay, this is reasonable and is how we handle the other probes. It would be nice to work towards limiting the edge cases we add to the schema generator, though.

Reply:
> It would be nice to work towards limiting the edge cases we add to the schema generator, though.

definitely